python/pyspark/pandas/frame.py

```diff
@@ -490,20 +491,30 @@ def _internal(self) -> InternalFrame:
         return cast(InternalFrame, self._internal_frame)  # type: ignore[has-type]
 
     def _update_internal_frame(
-        self, internal: InternalFrame, requires_same_anchor: bool = True
+        self,
+        internal: InternalFrame,
+        requires_same_anchor: bool = True,
+        force_disconnect: bool = False,
     ) -> None:
         """
         Update InternalFrame with the given one.
 
-        If the column_label is changed or the new InternalFrame is not the same `anchor`,
-        disconnect the link to the Series and create a new one.
+        If the column_label is changed or the new InternalFrame is not the same `anchor` or the
+        `force_disconnect` flag is set to True, disconnect the link to the Series and create a new
+        one.
```

Comment:
What if `requires_same_anchor` and `force_disconnect` are both `True` ?

Reply:
Oh, sorry. approve is a mistake.

But looks pretty good otherwise.

Reply:
Thanks for review, not sure get your idea completely.

It's A or B or C, Create new dataframe if one of three coditions statified, so `if requires_same_anchor and force_disconnect are both True` also create the new dataframe.

Reply:
Ah,,, I got it.

When I first see the name `requires_same_anchor`, I thought it means they have must same anchor between old and new `InternalFrame`, but seems like it's just for checking if they have same anchor ??

If so, how about rename the parameter to `check_same_anchor` or `verify_same_anchor` or something ?

I'm not very strong feeling about this, but I think maybe it's more proper naming.

WDYT?

Reply:
Yep, good suggestion, but `required_same_anchor` is orignal parameter, and already being used in current code.

Consider it's inertnal function, we could change it. I can submit a separate PR to address it to make PR easy to review.

Reply:
https://issues.apache.org/jira/browse/SPARK-39310 create a jira